zkp-component/src/proof_gen_utils.rs

```diff
@@ -78,9 +78,12 @@ pub async fn launch_generate_new_proof(
     };
     log::debug!("Launching the prover process at path {:?}", path);
 
+    let mut args = env::args().collect::<Vec<String>>();
+    args.remove(0);
+
     let mut child = Command::new(path)
         .arg("--prove")
-        .args(env::args())
+        .args(args)
```

Comment:
This should work equally well and avoids `collect`ing the vec :)
```suggestion
        .args(env::args().skip(1))
```